src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs

```diff
@@ -67,11 +71,23 @@ protected virtual void OnChildrenChanged()
 			{
 				case SolidColorBrush scb:
 					svgElement.SetStyle("fill", scb.Color.ToCssString());
+					_fillBrushSubscription.Disposable = null;
 					break;
 				case ImageBrush ib:
+					_fillBrushSubscription.Disposable = null;
+					break;
+				case LinearGradientBrush lgb:
+					var linearGradient = lgb.ToSvgElement();
+					var gradientId = linearGradient.HtmlId;
+					SvgChildren.Add(linearGradient);
+					svgElement.SetStyle("fill", $"url(#{gradientId})");
+					_fillBrushSubscription.Disposable = new DisposableAction(
+						() => SvgChildren.Remove(linearGradient)
+					);
```

Comment:
You should also support "Stroke", not just "Fill".

Reply:
Done!

I noticed that the Stroke widths were off, but this seems to be a known issue (https://github.com/unoplatform/uno/issues/1552).